docs/tutorials/infrastructure/kne/srl-with-kne/srl-with-oc-services.md

```diff
@@ -0,0 +1,255 @@
+
+    admin@172.18.0.50's password:
+    ```
+
+### Console
+
+To get a console-like access to SR Linux NOS users should leverage `kubectl exec` command and start the `sr_cli` process:
+
+```shell
+kubectl -n 2-srl-ixr6 exec -it srl1 -- sr_cli # (1)!
+```
+
+1. * Namespace `2-srl-ixr6` matches the lab name set in the topology file  
+    * `srl1` container name matches the node name set in the topology file.
+
+??? "Example"
+    ```
+    ❯ kubectl -n 2-srl-ixr6 exec -it srl1 -- sr_cli
+    Defaulted container "srl1" out of: srl1, init-srl1 (init)
+    Using configuration file(s): ['/etc/opt/srlinux/srlinux.rc']
+    Welcome to the srlinux CLI.
+    Type 'help' (and press <ENTER>) if you need any help using this.
+    --{ running }--[  ]--
+    A:srl1#
+    ```
+
+### Openconfig
+
+<small>[:octicons-book-16: Openconfig docs][oc-doc]</small>
+
+[oc-doc]: https://documentation.nokia.com/srlinux/22-6/SR_Linux_Book_Files/SysMgmt_Guide/data-models.html#openconfig-ov
+
+By default, Nokia SR Linux uses native [YANG models](../../../../yang/yang.md). Openconfig YANG models are already enabled in the configuration file used in this lab.
+
+For completeness, the below section shows how to enable Openconfig via different management interfaces.
+
+=== "CLI"
+    ```
+    --{ running }--[  ]--
+    A:srl# enter candidate
+
+    --{ candidate shared default }--[  ]--
+    A:srl# system management openconfig admin-state enable
+    
+    --{ * candidate shared default }--[  ]--
+    A:srl# commit stay 
+    All changes have been committed. Starting new transaction.
+    ```
+=== "Config file"
+    ```json
+    "srl_nokia-system:system": {
+      "management": {
+        "srl_nokia-openconfig:openconfig": {
+          "admin-state": "enable"
+        }
+      },
+    // other system containers
+    }
+    ```
+
+### gNMI
+
+<small>[:octicons-book-16: gNMI docs][gnmi-doc]</small>
+
+[gnmi-doc]: https://documentation.nokia.com/srlinux/22-6/SR_Linux_Book_Files/SysMgmt_Guide/gnmi-interface.html
+
+gNMI service is enabled over port `57400` in the configuration files used with this lab.
+
+<!-- For completeness, the below section shows how to enable gNMI via different management interfaces.
+
+=== "CLI"
+    ```
+    --{ running }--[  ]--
+    A:srl# enter candidate
+
+    --{ candidate shared default }--[  ]--
+    A:srl# system management openconfig admin-state enable
+    
+    --{ * candidate shared default }--[  ]--
+    A:srl# commit stay 
+    All changes have been committed. Starting new transaction.
+    ```
+=== "Config file"
+    ```json
+    "srl_nokia-system:system": {
+      "management": {
+        "srl_nokia-openconfig:openconfig": {
+          "admin-state": "enable"
+        }
+      },
+    // other system containers
+    }
+    ``` -->
+
+??? "Example"
+    gNMI service can be tested using [gnmic](https://gnmic.kmrd.dev) cli client.
+
+    ```bash
+    ❯ gnmic -a 172.18.0.50:57400 -u admin -p admin --skip-verify capabilities
```

Comment:
port 9339 with the topo proto changes

Reply:
also getting an error:

```
alexmasi@alexmasi:~$ kubectl get services -A
NAMESPACE            NAME                                                    TYPE           CLUSTER-IP      EXTERNAL-IP     PORT(S)                                                                    AGE
2-srl-ixr6           service-srl1                                            LoadBalancer   10.96.207.160   192.168.11.50   9339:30295/TCP,9340:32633/TCP,9559:32639/TCP,22:31363/TCP,9337:32505/TCP   86s
2-srl-ixr6           service-srl2                                            LoadBalancer   10.96.138.80    192.168.11.51   9339:32597/TCP,9340:31831/TCP,9559:32579/TCP,22:32581/TCP,9337:31614/TCP   86s
default              kubernetes                                              ClusterIP      10.96.0.1       <none>          443/TCP                                                                    3m21s
kube-system          kube-dns                                                ClusterIP      10.96.0.10      <none>          53/UDP,53/TCP,9153/TCP                                                     3m19s
metallb-system       webhook-service                                         ClusterIP      10.96.244.209   <none>          443/TCP                                                                    3m16s
srlinux-controller   srlinux-controller-controller-manager-metrics-service   ClusterIP      10.96.0.194     <none>          8443/TCP                                                                   2m8s
alexmasi@alexmasi:~$ gnmic -a 192.168.11.50:9339 -u admin -p admin --skip-verify capabilities
target "192.168.11.50:9339", capabilities request failed: failed to create a gRPC client for target '192.168.11.50:9339' : 192.168.11.50:9339: context deadline exceeded
Error: one or more requests failed
```

Reply:
I think I haven't pushed the changes to the repo and only updated the staging server. Pushed it now.

On my system I do not see the issue with gnmi/gnmic:

```
❯ gnmic -a 172.18.0.50:9339 -u admin -p admin --skip-verify capabilities
gNMI version: 0.7.0
supported models:
  - urn:srl_nokia/aaa:srl_nokia-aaa, Nokia, 2022-06-30
  - urn:srl_nokia/aaa-password:srl_nokia-aaa-password, Nokia, 2022-06-30
```

can you ensure that your pods are running fine?

```
kubectl get pods -n 2-srl-ixr6 
```

Reply:
yeah its in a crashloopbackoff loop, I will check tomorrow exactly when that occurs, here are the logs:

```
$ kubectl logs srl1 -n 2-srl-ixr6
Defaulted container "srl1" out of: srl1, init-srl1 (init)
topomac.sh script finished
Fri Oct  7 07:26:57 UTC 2022: entrypoint.sh called
Fri Oct  7 07:26:57 UTC 2022: renaming docker interface eth0 to mgmt0
Cannot find device "eth0"
Device "eth0" does not exist.
Cannot find device "eth0"
Cannot find device "eth0"
Fri Oct  7 07:26:58 UTC 2022: turning off checksum offloading on mgmt0
Fri Oct  7 07:26:58 UTC 2022: starting sshd
ssh-keygen: generating new host keys: RSA DSA ECDSA ED25519
Fri Oct  7 07:26:58 UTC 2022: Calling boot_run script
/opt/srlinux/bin/bootscript/05_sr_createuser.sh: line 270: !srl_is_running_on_nokia_rootfs: command not found
/opt/srlinux/bin/bootscript/05_sr_createuser.sh: line 282: python: command not found
/opt/srlinux/bin/logmgr_set_env.sh: line 480: /sys/fs/cgroup/user.slice/cgroup.procs: No such file or directory
logmgr_set_env.sh: setcap for sr_supportd => 0
chmod: cannot access '/dev/console': No such file or directory
chmod: missing operand after '0664'
Try 'chmod --help' for more information.
/usr/bin/find: '/var/log/srlinux/file': No such file or directory
logmgr_set_env.sh: plain_bootup_start
Fri Oct  7 07:27:01 UTC 2022  logmgr_set_env.sh: restart of rsyslogd
setfacl: /mnt/nokiaos: No such file or directory
setfacl: /mnt/nokiaos: No such file or directory
setfacl: /mnt/nokiaos: No such file or directory
setfacl: /mnt/nokiaos: No such file or directory
setfacl: Option -m: Invalid argument near character 5
setfacl: Option -m: Invalid argument near character 3
setfacl: Option -m: Invalid argument near character 5
setfacl: Option -m: Invalid argument near character 3
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
Failed to open connection to "system" message bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
System has not been booted with systemd as init system (PID 1). Can't operate.
Failed to connect to bus: Host is down
findfs: unable to resolve 'LABEL=EFI-System'
No disk with label EFI-System is found
Failed to set capabilities on file `/usr/sbin/tcpdump' (No such file or directory)
usage: setcap [-h] [-q] [-v] [-n <rootid>] (-r|-|<caps>) <filename> [ ... (-r|-|<capsN>) <filenameN> ]

 Note <filename> must be a regular (non-symlink) file.
 -r          remove capability from file
 -           read capability text from stdin
 <capsN>     cap_from_text(3) formatted file capability

 -h          this message and exit status 0
 -q          quietly
 -v          validate supplied capability matches file
 -n <rootid> write a user namespace limited capability
 --license   display the license info
Fri Oct  7 07:27:03 UTC 2022: entrypoint.sh done, executing sudo bash -c touch /.dockerenv && /opt/srlinux/bin/sr_linux
No/Invalid license found!
Not starting in a named namespace, giving it the name "srbase"
Unix domain socket directory is /opt/srlinux/var/run/
Log directory is /var/log/srlinux/stdout
  Started supportd: source /etc/profile.d/sr_app_env.sh &>/dev/null; bash -c "./sr_supportd --server-mode" >/var/log/srlinux/stdout/supportd.log 2>&1 &
  Application supportd is running: PID 1555
  Started dev_mgr: source /etc/profile.d/sr_app_env.sh &>/dev/null; bash -c "./sr_device_mgr" >/var/log/srlinux/stdout/dev_mgr.log 2>&1 &
  Application dev_mgr is running: PID 1576
  Found dev_mgr: PID 1576 - killing
  Found supportd: PID 1555 - killing
  Failed to kill supportd: PID 1555. Another kill is retried
```

Reply:
```
No/Invalid license found!
```

there was no license found by the pod

Reply:
@alexmasi have you been able to get them running? Any help needed from my side?

Reply:
Sorry had other tasks getting in the way. Just tried again and verified the licenses were set up as per the intructions in srl-controller repo. However still getting the same issue when I tried again.

The license file ends in `# master`, is this a wildcard to catch all versions or should I change it to a specific version to match the container?

Reply:
@alexmasi yes, the lic you have (if it doesn't have other lic blobs inside) is supposed to work with the build we sent you. Can you change the image to 0.0.0-something matching the image that you have gotten from us?

Reply:
I tried using the tag `# srl_rel_22_06_*` instead of `# master` in the license file as am I using the public image specified in the topo: `ghcr.io/nokia/srlinux:22.6.4`

However I still get the same failure

Does this require the internal release you shared with us? If so I will try again with that image

Reply:
> Does this require the internal release you shared with us? If so I will try again with that image

yes, that is the only combination that would work

Reply:
Just tried with the internal image and both `# master` and `0.0.0-<omitted>` and both times the creation hangs with:

```
INFO[0000] Adding Link: srl1:e1-1 srl2:e1-1             
INFO[0000] Adding Node: srl1:NOKIA:UNKNOWN              
INFO[0000] Adding Node: srl2:NOKIA:UNKNOWN              
INFO[0000] Created manager for topology:
...
INFO[0000] Creating namespace for topology: "2-srl-ixr6" 
INFO[0000] Server Namespace: &Namespace{ObjectMeta:{2-srl-ixr6    c6b51bac-a6c2-4255-8dbb-7143fd3969c6 1006093 0 2022-10-11 17:17:19 -0700 PDT <nil> <nil> map[kubernetes.io/metadata.name:2-srl-ixr6] map[] [] []  [{kne_cli Update v1 2022-10-11 17:17:19 -0700 PDT FieldsV1 {"f:metadata":{"f:labels":{".":{},"f:kubernetes.io/metadata.name":{}}}} }]},Spec:NamespaceSpec{Finalizers:[kubernetes],},Status:NamespaceStatus{Phase:Active,Conditions:[]NamespaceCondition{},},} 
INFO[0000] Getting topology specs for namespace 2-srl-ixr6 
INFO[0000] Getting topology specs for node srl1         
INFO[0000] Getting topology specs for node srl2         
INFO[0000] Creating topology for meshnet node srl1      
INFO[0000] Meshnet Node:
&{TypeMeta:{Kind:Topology APIVersion:networkop.co.uk/v1beta1} ObjectMeta:{Name:srl1 GenerateName: Namespace:2-srl-ixr6 SelfLink: UID:5c5f7eb2-78e1-4286-a6f0-f3ad3693d0aa ResourceVersion:1006094 Generation:1 CreationTimestamp:2022-10-11 17:17:19 -0700 PDT DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ZZZ_DeprecatedClusterName: ManagedFields:[{Manager:kne_cli Operation:Update APIVersion:networkop.co.uk/v1beta1 Time:2022-10-11 17:17:19 -0700 PDT FieldsType:FieldsV1 FieldsV1:{"f:spec":{".":{},"f:links":{}}} Subresource:}]} Status:{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name: GenerateName: Namespace: SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ZZZ_DeprecatedClusterName: ManagedFields:[]} Skipped:[] SrcIP: NetNS:} Spec:{TypeMeta:{Kind: APIVersion:} Links:[{LocalIntf:e1-1 LocalIP: PeerIntf:e1-1 PeerIP: PeerPod:srl2 UID:0}]}} 
INFO[0000] Creating topology for meshnet node srl2      
INFO[0000] Meshnet Node:
&{TypeMeta:{Kind:Topology APIVersion:networkop.co.uk/v1beta1} ObjectMeta:{Name:srl2 GenerateName: Namespace:2-srl-ixr6 SelfLink: UID:d9298ee8-5a19-4756-bdcb-e3f11ca08af4 ResourceVersion:1006097 Generation:1 CreationTimestamp:2022-10-11 17:17:19 -0700 PDT DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ZZZ_DeprecatedClusterName: ManagedFields:[{Manager:kne_cli Operation:Update APIVersion:networkop.co.uk/v1beta1 Time:2022-10-11 17:17:19 -0700 PDT FieldsType:FieldsV1 FieldsV1:{"f:spec":{".":{},"f:links":{}}} Subresource:}]} Status:{TypeMeta:{Kind: APIVersion:} ObjectMeta:{Name: GenerateName: Namespace: SelfLink: UID: ResourceVersion: Generation:0 CreationTimestamp:0001-01-01 00:00:00 +0000 UTC DeletionTimestamp:<nil> DeletionGracePeriodSeconds:<nil> Labels:map[] Annotations:map[] OwnerReferences:[] Finalizers:[] ZZZ_DeprecatedClusterName: ManagedFields:[]} Skipped:[] SrcIP: NetNS:} Spec:{TypeMeta:{Kind: APIVersion:} Links:[{LocalIntf:e1-1 LocalIP: PeerIntf:e1-1 PeerIP: PeerPod:srl1 UID:0}]}} 
INFO[0000] Creating Node Pods                           
INFO[0000] Creating Srlinux node resource srl1          
INFO[0000] Server Config Map:
&ConfigMap{ObjectMeta:{srl1-config  2-srl-ixr6  fbd1a465-7a02-45bb-90c5-2c6b50689fcf 1006098 0 2022-10-11 17:17:19 -0700 PDT <nil> <nil> map[] map[] [] []  [{kne_cli Update v1 2022-10-11 17:17:19 -0700 PDT FieldsV1 {"f:data":{".":{},"f:config.json":{}}} }]},Data:map[string]string{config.json: {
  "_preamble": {
    "header": {
     ...
INFO[0000] Created SR Linux node srl1 configmap
```

and the pods never appearing. This is our first time trying the internal container version so this may not be related to the licensing

Reply:
this doesn't sound like something related to the image and/or licenses

Reply:
a suggestion would be to doublecheck that you're not using the old srl-controller that kne manifests define.

remove the old ones with:
```
kubectl delete -k https://github.com/srl-labs/srl-controller/config/default
```

and install the latest with:
```
kubectl apply -k https://github.com/srl-labs/srl-controller/config/default
```

Reply:
Yes I am running with 0.4.4

Reply:
@alexmasi Yes, it was tested
I just tested it myself and it starts up just fine.

I used the latest drop we provided to you, which was 
```
Software Version     : v0.0.0
Build Number         : 38011-g0eaf8a56cc
```

and used a license file available in the same share